backend/session/session.c

```diff
@@ -533,3 +533,59 @@ ssize_t wlr_session_find_gpus(struct wlr_session *session,
 
 	return i;
 }
+
+struct wait_session_active_handler {
+	bool active;
+	struct wl_listener listener;
+};
+
+static void	wait_session_active(struct wl_listener *listener, void *data) {
+	struct wait_session_active_handler *handler =
+		wl_container_of(listener, handler, listener);
+	handler->active = true;
+}
+
+struct wlr_session *wlr_session_create(struct wl_display *disp) {
```

Comment:
I'd rather do this only when `wlr_session_create` is called from `wlr_backend_autocreate`. This gives compositors a chance to not block when VT switched away at startup if they want to.

Reply:
@emersion, could you offer name for new function to be exported then? `wlr_session_create_blocking` or `wlr_session_create_waiting` maybe.

Reply:
Leave `wlr_session_create` as-is, and introduce a new static function in `backend/backend.c` named e.g. `create_and_wait_session`?

Reply:
also there are two places, one from `wlr_backend_autocreate` the other in `attempt_backend_by_name` in `backend/backend.c`. where `attempt_backend_by_name` is also called from `wlr_backend_autocreate`. `ripgrep` don't report any other usages, that is why i decided not to change the name.